### PR TITLE
OKTA-543013 Add Login.gov social IdP

### PR DESCRIPTION
**Pull request by vanngo-okta**
## Description:
- **What's changed?** Add Login.gov Production and Sandbox social IdP guide
- **Is this PR related to a Monolith release?** Yes, 2023.02.0 (see OKTA-570479)

### Previews:
* (Feb 6, 2023, removed Private Key EA flag) https://63e18adca0965b22c6a0a1c6--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (Nov 17 - after editorial review) https://6375b7ffedb37f1cadaaf002--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (draft 4 - Nov 10 - Reorg Test section) https://636d7fbe45b2ff44c183dbc0--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (draft 3 - Nov 10) https://636d292aff69b552d8a53e8b--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (draft 2 - Nov 9) https://636be394d27cca0086b2d4f0--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (draft 1 - Nov 9) https://636bbe853dade47539723da7--reverent-murdock-829d24.netlify.app/docs/guides/add-logingov-idp/main/
* (draft social IdP version) https://6363ebb547fc0e0077c35524--reverent-murdock-829d24.netlify.app/docs/guides/identity-providers/

### Resolves:

* [OKTA-543013](https://oktainc.atlassian.net/browse/OKTA-543013)

### CODE REVIEW

**Comment by johnbaldo-okta** [comment]
Look good from my perspective! A couple thoughts that came up towards the end of the page...
* @ethanhsiung-okta , did we add this to the Sign-in widget? I'm having trouble remembering, but I thought we had to manually add them. I know we had to add the icons at least. If that's the case, I'm worried people would get an error.
* For testing / integrating... with OIE, I thought the routing rule was the easiest way to go and it just gets added to the hosted sign in widget automatically. If that's the easiest and preferred thing, is there any way we can steer people in that direction? Anything we do for other IdPs? It's just the one sentence and then most of the content is around other methods, so I thought it might be easy to miss.

**Comment by vanngo-okta** [comment]
Joel Franusic (CIAM specialist) reviewed and approved on Slack ✅

**Review by johnbaldo-okta — APPROVED** [review]

**Comment by ethanhsiung-okta** [comment]
> * @ethanhsiung-okta , did we add this to the Sign-in widget? I'm having trouble remembering, but I thought we had to manually add them. I know we had to add the icons at least. If that's the case, I'm worried people would get an error.

@johnbaldo-okta I didn't add this to the sign-in widget. It wasn't mentioned in Cole's wiki https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2271547222/How+to+Configure+Metadata-Driven+Social+Identity+Providers+Metadata-Driven+Social+IdPs , but I'll look around

**Review by ethanhsiung-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    * **Default Authentication Assurance Level (AAL)**: Select AAL required. Ensure that this value matches the **AAL value** specified in [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).
+       > **Note:** If you leave the default value, Login.gov requires a user to be authenticated with a second factor: `urn:gov:gsa:ac:classes:sp:PasswordProtectedTransport:duo`. See [AAL Values](https://developers.login.gov/oidc/#aal-values).
+    * **Attribute bundle**: Select the default attributes for Login.gov to return during authentication. Select at least **email**, **profile**, and **profile:email**.
+    * **Issuer**: Specify a string that matches the **Client ID** value in your Okta Login.gov IdP configuration from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta). Typically, your Okta org URL is used. For example, if your Okta subdomain is `company`, then specify `https://company.okta.com`.
+    * **Certificates**: Upload the public key file you downloaded from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:104]:
```suggestion
    * **Certificates**: Upload the public key file that you downloaded from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    * **Redirect URIs**: Specify your Okta **Redirect URI** you copied from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta). For example, if you don't have a custom domain, this value is usually `https://${yourCompanySubdomain}.okta.com/oauth2/v1/authorize/callback`.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:105]:
```suggestion
    * **Redirect URIs**: Specify the Okta **Redirect URI** that you copied from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta). For example, if you don't have a custom domain, this value is usually `https://${yourCompanySubdomain}.okta.com/oauth2/v1/authorize/callback`.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+* [Use an embedded Okta Sign-In Widget](#use-an-embedded-okta-sign-in-widget)
+* [Use a custom Okta-hosted Sign-In Widget](#use-a-custom-okta-hosted-sign-in-widget)
+* [Use an HTML link](#use-an-html-link)
+* [Use the AuthJS SDK](#use-the-authjs-sdk)
+
+### Use an embedded Okta Sign-In Widget
+
+Okta also offers an easily embeddable JavaScript widget that reproduces the look and behavior of the standard Okta sign-in page for your client app. See [Embedded Sign-In Widget use cases](/docs/guides/embedded-sign-in-widget-use-cases/) for Identity Engine or [Embedded Okta Sign-In Widget fundamentals](/docs/guides/archive-embedded-siw/main/) for Classic Engine.
+
+To test Login.gov IdP sign-in with your embedded widget app, navigate to your app's sign-in page and click the **Sign in with Login.gov** option. This option directs you to the Login.gov sign-in page. Use your Login.gov credentials to test the sign-in flow.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:126]:
```suggestion
To test the Login.gov IdP sign-in flow with your embedded widget app, navigate to your app's sign-in page and click the **Sign in with Login.gov** option. This option directs you to the Login.gov sign-in page. Use your Login.gov credentials to test the sign-in flow.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+There are four ways to initiate the IdP sign-in flow:
+
+* [Use an embedded Okta Sign-In Widget](#use-an-embedded-okta-sign-in-widget)
+* [Use a custom Okta-hosted Sign-In Widget](#use-a-custom-okta-hosted-sign-in-widget)
+* [Use an HTML link](#use-an-html-link)
+* [Use the AuthJS SDK](#use-the-authjs-sdk)
+
+### Use an embedded Okta Sign-In Widget
+
+Okta also offers an easily embeddable JavaScript widget that reproduces the look and behavior of the standard Okta sign-in page for your client app. See [Embedded Sign-In Widget use cases](/docs/guides/embedded-sign-in-widget-use-cases/) for Identity Engine or [Embedded Okta Sign-In Widget fundamentals](/docs/guides/archive-embedded-siw/main/) for Classic Engine.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:124]:
```suggestion
Okta also offers an easily embeddable JavaScript widget that reproduces the look and behavior of the standard Okta sign-in page for your client app. See [Embedded Sign-In Widget use cases](/docs/guides/embedded-sign-in-widget-use-cases/) for Okta Identity Engine or [Embedded Okta Sign-In Widget fundamentals](/docs/guides/archive-embedded-siw/main/) for Okta Classic Engine.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+
+To test Login.gov IdP sign-in with your embedded widget app, navigate to your app's sign-in page and click the **Sign in with Login.gov** option. This option directs you to the Login.gov sign-in page. Use your Login.gov credentials to test the sign-in flow.
+
+#### Embedded widget with Identity Engine
+
+If you're using Okta Identity Engine, the **Sign in with Login.gov** option is available on the widget after you [create an Identity Provider in your Okta org](#create-an-identity-provider-in-okta) and configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). No additional code is required.
+
+#### Embedded widget with Classic Engine
+
+If you're using Okta Classic Engine, you still need to configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). In addition, you need to add the following code to your Okta Sign-In Widget configuration to display the **Sign in with Login.gov** button:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:134]:
```suggestion
If you're using Classic Engine, you still need to configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). Additionally, you need to add the following code to your Okta Sign-In Widget configuration to display the **Sign in with Login.gov** button:
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+### Use an embedded Okta Sign-In Widget
+
+Okta also offers an easily embeddable JavaScript widget that reproduces the look and behavior of the standard Okta sign-in page for your client app. See [Embedded Sign-In Widget use cases](/docs/guides/embedded-sign-in-widget-use-cases/) for Identity Engine or [Embedded Okta Sign-In Widget fundamentals](/docs/guides/archive-embedded-siw/main/) for Classic Engine.
+
+To test Login.gov IdP sign-in with your embedded widget app, navigate to your app's sign-in page and click the **Sign in with Login.gov** option. This option directs you to the Login.gov sign-in page. Use your Login.gov credentials to test the sign-in flow.
+
+#### Embedded widget with Identity Engine
+
+If you're using Okta Identity Engine, the **Sign in with Login.gov** option is available on the widget after you [create an Identity Provider in your Okta org](#create-an-identity-provider-in-okta) and configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). No additional code is required.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:130]:
```suggestion
If you're using Identity Engine, the **Sign in with Login.gov** option is available on the widget after you [create an Identity Provider in your Okta org](#create-an-identity-provider-in-okta) and configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). No additional code is required.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+If you're using Okta Classic Engine, you still need to configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). In addition, you need to add the following code to your Okta Sign-In Widget configuration to display the **Sign in with Login.gov** button:
+
+```javascript
+    config.idps= [
+        { type: 'LOGINGOV_SANDBOX', id: 'Your_IDP_ID' }
+    ];
+    config.idpDisplay = "SECONDARY";
+```
+
+You can find out more about the Okta Sign-In Widget [on GitHub](https://github.com/okta/okta-signin-widget#okta-sign-in-widget). Implementing sign in with an Identity Provider uses the Widget's [OpenID Connect authentication flow](https://github.com/okta/okta-signin-widget#openid-connect).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:143]:
```suggestion
You can find out more about the Okta Sign-In Widget [on GitHub](https://github.com/okta/okta-signin-widget#okta-sign-in-widget). Implementing sign-in flows with an Identity Provider uses the Widget's [OpenID Connect authentication flow](https://github.com/okta/okta-signin-widget#openid-connect).
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+
+> **Note:** For production environment, use the following Okta Sign-In Widget configuration:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:145]:
```suggestion
> **Note:** For a production environment, use the following Okta Sign-In Widget configuration:
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+>    config.idpDisplay = "SECONDARY";
+>```
+
+### Use a custom Okta-hosted Sign-In Widget
+
+If your client app uses Okta [redirect authentication](#/docs/concepts/redirect-vs-embedded/#redirect-authentication), then to test the Login.gov IdP sign-in flow, navigate to your app's sign-in page and click the **Sign in with Login.gov** option. This option directs you to the Login.gov sign-in page. Use your Login.gov credentials to test the sign-in flow.
+
+#### Okta-hosted widget with Identity Engine
+
+If you're using Okta Identity Engine, the **Sign in with Login.gov** option is available on the widget after you [create an Identity Provider in your Okta org](#create-an-identity-provider-in-okta) and configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). No additional configuration is required.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:159]:
```suggestion
If you're using Identity Engine, the **Sign in with Login.gov** option is available on the widget after you [create an Identity Provider in your Okta org](#create-an-identity-provider-in-okta) and configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). No additional configuration is required.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+
+#### Okta-hosted widget with Classic Engine
+
+If you're using Okta Classic Engine, you still need to configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). In addition, you need to customize your [Okta-hosted Sign-In Widget](/docs/guides/custom-widget/main/#style-the-okta-hosted-sign-in-widget) to display the **Sign in with Login.gov** option.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:163]:
```suggestion
If you're using Classic Engine, you still need to configure the [routing rule](https://help.okta.com/okta_help.htm?id=ext-cfg-routing-rules). Additionally, you need to customize your [Okta-hosted Sign-In Widget](/docs/guides/custom-widget/main/#style-the-okta-hosted-sign-in-widget) to display the **Sign in with Login.gov** option.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+Add the following code beneath the `var config = OktaUtil.getSignInWidgetConfig();` line in the **Custom Sign in** tab:
+
+```js
+config.idps= [
+  {type: 'LOGINGOV_SANDBOX', id: 'Your_IDP_ID'}
+];
+config.idpDisplay ="SECONDARY";
+```
+
+> **Note:** For production environment, use the following code beneath the `var config = OktaUtil.getSignInWidgetConfig();` line:
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:174]:
```suggestion
> **Note:** For a production environment, use the following code beneath the `var config = OktaUtil.getSignInWidgetConfig();` line:
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+  redirect_uri=${redirectUri}&
+  state=${state}&
+  nonce=${nonce}
+```
+
+> **Note:** If you are using Authorization Code with PKCE as the grant type for your client app, you must generate and store the PKCE. See [Implement authorization by grant type](/docs/guides/implement-grant-type/authcodepkce/main/#flow-specifics). Okta recommends that you use the [AuthJS SDK](https://github.com/okta/okta-auth-js#signinwithredirectoptions) with this grant type.
+
+In the URL, replace `${yourOktaDomain}` with your org's base URL, and then replace the following values:
+
+* `idp`: Your `${yourIdPId}` value you obtained from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:210]:
```suggestion
* `idp`: Your `${yourIdPId}` value that you obtained from [Create an Identity Provider in Okta](#create-an-identity-provider-in-okta).
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+
+* `client_id`: Use the `${clientId}` value that you obtained from your OpenID Connect client application. This is not the **Client ID** value from the Identity Provider.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:212]:
```suggestion
* `client_id`: Use the `${clientId}` value that you obtained from your OpenID Connect client application. This isn't the **Client ID** value from the Identity Provider.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+
+* `scope`: Determines the claims that are returned in the ID token. Include the scopes that you want to request authorization for and separate each by a space. You need to include at least the `openid` scope. You can request any of the standard OpenID Connect scopes about users, such as `profile` and `email`, as well as any custom scopes specific to your Identity Provider.
+
+* `redirect_uri`: The location where Okta returns a browser after the user finishes authenticating with their Identity Provider. This URL must start with `https` and must match one of the redirect URIs that you configured in the previous section.
+
+* `state`: Protects against cross-site request forgery (CSRF). Can be any value.
+
+* `nonce`: A string included in the returned ID token. Use it to associate a client session with an ID token and to mitigate replay attacks. Can be any value.
+
+> **Note:** When Okta sends the authorize request to Login.gov, it adds the required `acr_values` based on the type of identity verification you specified in the Login.gov IdP configuration.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:226]:
```suggestion
> **Note:** When Okta sends the authorize request to Login.gov, it adds the required `acr_values` based on the type of identity verification that you specified in the Login.gov IdP configuration.
```

File: packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+```html
+`<a href="https://${yourOktaDomain}/oauth2/v1/authorize?idp=0oaaq9pjc2ujmFZexample&client_id=GkGw4K49N4UEE1example&response_type=id_token&response_mode=fragment&scope=openid&redirect_uri=${yourAppRedirectUrl}&state=WM6D&nonce=YsG76jo">Sign in with Login.gov</a>`
+
+```
+
+After the user clicks the link, they are prompted to sign in with Login.gov. After successful sign in, the user is returned to the specified `redirect_uri` along with an ID token in JWT format.
+
+### Use the AuthJS SDK
+
+If you don't want pre-built views, or need deeper levels of customization, then you can use the same AuthJS SDK that the Sign-in Widget is built with. See the `idp` option in the [AuthJS SDK GitHub repo](https://github.com/okta/okta-auth-js#usage-guide).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/add-logingov-idp/main/index.md:241]:
```suggestion
If you don't want pre-built views, or need deeper levels of customization, then you can use the same AuthJS SDK that the Sign-In Widget is built with. See the `idp` option in the [AuthJS SDK GitHub repo](https://github.com/okta/okta-auth-js#usage-guide).
```

**Review by susanharper-okta — APPROVED** [review]
couple style and other edits. approving to not hold you up.

**Comment by vanngo-okta** [comment]
Legal review approved by Stephanie Scafaria Nov 17, 2022 via Slack.

**Pull request merged**